exonum/src/events/noise/wrappers/sodium_wrapper/handshake.rs

```diff
@@ -143,35 +192,46 @@ impl NoiseHandshake {
 }
 
 impl Handshake for NoiseHandshake {
-    fn listen<S>(self, stream: S) -> HandshakeResult<S>
+    type Result = Vec<u8>;
```

Comment:
Why is this type so abstract? Can't we return `SocketAddr` here? (For now; in general, we want to support domain names if I understand correctly.)